### ISSUE TITLE
Error on running app: Native runtime is not available and Error (Invariant Violation: Module AppRegistry)

### ISSUE THREAD

**Issue by garethf89** [expo/expo#20249]
### Summary

I am trying to build an app with Expo for the first time and I have encountered some errors that is preventing me from loading an app with Expo Go.

I have downloaded a starter package to debug and all I see is 'Expo native runtime is not available'. This is using the official Turborepo example here https://github.com/vercel/turbo/tree/main/examples/with-react-native-web (I get the same when extracting the app out of the monorepo)

![Screenshot 2022-11-28 at 17 21 30](https://user-images.githubusercontent.com/5473451/204349135-bfaf1978-f954-4bf0-bf19-8dd2ea23caca.png)

On my own app that I am developing, I get the same error in addition to many more:

`app:dev: Logs for your project will appear below.`
`app:dev: Android Bundling complete 13937ms`
`app:dev:  WARN  The "EXNativeModulesProxy" native module is not exported through NativeModules; verify that expo-modules-core's native code is linked properly`
`app:dev:  WARN  No native ExponentConstants module found, are you sure the expo-constants's module is linked properly?`
`app:dev:  ERROR  Error: Expo native runtime is not available: something went wrong and we aren't sure what it was. Please post more information and get support at https://forums.expo.dev.`
`app:dev:  ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.`
`app:dev:       This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.`
`app:dev:  ERROR  Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.`
`app:dev:       This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native.`

Again I am not really doing much with this app other than a few basic screens.

This works fine when developing using Web and it only does this when I try to use a device using Expo Go.

Any help with this would be appreciated as I have tried everything to resolve this, thanks!

### What platform(s) does this occur on?

Android

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.0.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
      Yarn: 3.2.4 - /usr/local/bin/yarn
      npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
      Watchman: 2022.10.03.00 - /usr/local/bin/watchman
    IDEs:
      Xcode: /undefined - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 6.0.6
    Expo Workflow: managed

### Minimal reproducible example

This is happening with every app but this is an example of one:

https://github.com/vercel/turbo/tree/main/examples/with-react-native-web

**Comment by brentvatne** [comment]
hi there! this issue seems better suited for the vercel/turbo repository, we do not maintain that example.

**Comment by garethf89** [comment]
@brentvatne This issue happens with any application, that was just an example that is easy to share.

**Comment by garethf89** [comment]
I have been able to fix this issue now, it was a problem with my Babel config. Thanks for the response, I am happy for it to be closed now.

**Comment by brentvatne** [comment]
good to hear! that's exactly why we ask for a reproducible example - usually folks find the root cause in the process of making one, and without it, it's very tough for us to do it

**Comment by mikevocalz** [comment]
> I have been able to fix this issue now, it was a problem with my Babel config. Thanks for the response, I am happy for it to be closed now.

how did you wind up fixing it? @brentvatne

**Comment by ClaudionorJunior** [comment]
@garethf89 Can you describe further what was the cause of the problem and where exactly in Babel was the problem?

**Comment by hesingon** [comment]
> 

How did you fix it? I'm stuck with the same issue

**Comment by eugali** [comment]
same

**Comment by JustinmClapperton** [comment]
Please share how you fixed it

**Comment by garethf89** [comment]
Sorry for the delay in response here, I missed the notifications.

There were two things I did to fix this issue and other similar errors.

1. In my app where I had this issue I was using a Monorepo folder structure, it was causing me to have a lot of issues like this. I switched to using installConfig.hoistingLimits limits on Yarn 3 to ensure all the apps modules are installed in a local node_modules folder and not a global shared one. (I belive this is the same as 'nohoist' in classic yarn)

2. Removed react-native-web from the babel config. Specifically this line in plugins `["react-native-web", { commonjs: true }],`

I am happy to provide any more specific information about my setup if it's useful.

My `babel.config.js` is super simple now

```
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ["babel-preset-expo"],
    plugins: [
      "react-native-reanimated/plugin",
    ],
  };
};
```
